packages/core/utils/src/event-bus/index.ts

```diff
@@ -85,12 +85,14 @@ export abstract class AbstractEventBusModuleService
      * otherwise we generate a random using a ulid
      */
 
-    const randId = ulid()
     const event = eventName.toString()
+    const subscriberId = context?.subscriberId ?? `${event}-${ulid()}`
+
+    ;(subscriber as any).subscriberId = subscriberId
```

Comment:
In which cases would the context be empty? Mutating the `subscriber` is maybe not the best, but I guess at least we have tests to ensure this keeps working in the future